src/pytest_bdd/scenario.py

```diff
@@ -57,6 +58,10 @@
 
 scenario_wrapper_template_registry: WeakKeyDictionary[Callable[..., object], ScenarioTemplate] = WeakKeyDictionary()
 
+# HACK: call_fixture_func will appear in the traceback whenever a step fails. This is
+# ugly and irrelevant, so this hack will hide it.
+_pytest.fixtures.__tracebackhide__ = True
```

Comment:
we should not patch pytest internals

Reply:
arguably we shouldn't be *using* pytest internals either, but point taken. I can rework it to not do that, though it's a bit more complicated.